src/applications/disability-benefits/wizard/pages/bdd.jsx

```diff
@@ -55,76 +50,68 @@ const findNextPage = state => {
   return pageNames.unableToFileBDD;
 };
 
-const defaultState = {
-  day: {
-    value: '',
-    dirty: false,
-  },
-  month: {
-    value: '',
-    dirty: false,
-  },
-  year: {
-    value: '',
-    dirty: false,
-  },
-};
-
 const label =
   'What’s the date or anticipated date of your release from active duty?';
 
-const isDateComplete = date =>
-  date.day.value && date.month.value && date.year.value.length === 4;
-
-const getDate = date =>
-  moment({
-    day: date.day.value,
-    month: parseInt(date.month.value, 10) - 1,
-    year: date.year.value,
-  });
-
+const getDate = date => moment(date, dateTemplate);
+const isDateComplete = date => date?.length === dateTemplate.length;
 const isDateInFuture = date => date?.diff(moment()) > 0;
 const isDateLessThanMax = date => date?.isBefore(maxDate);
 
-const BDDPage = ({ setPageState, state = defaultState }) => {
+const BDDPage = ({ setPageState, state = '' }) => {
   const [ariaDescribedby, setAriaDescribedby] = useState('');
+  const [isDirty, setIsDirty] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
 
-  const onChange = pageState => {
+  const onChange = event => {
+    const pageState = event.target.value || '';
     saveDischargeDate();
     const date = isDateComplete(pageState) ? getDate(pageState) : null;
-    const value =
+    const nextPage =
       isDateInFuture(date) && isDateLessThanMax(date)
         ? findNextPage(pageState)
         : null;
 
     // invalid date & page
-    setPageState(pageState, date && value === null ? 1 : value);
-    if (date && value) {
+    setPageState(pageState, date && nextPage === null ? 1 : nextPage);
+
+    if (date && nextPage) {
       // only set when there's a valid date
-      setAriaDescribedby(value);
+      setAriaDescribedby(nextPage);
       recordEvent({
         event: 'howToWizard-formChange',
         // Date component wrapper class name
         'form-field-type': 'usa-date-of-birth',
         'form-field-label': label,
-        'form-field-value': date.format('YYYY-MM-DD'),
+        'form-field-value': pageState,
       });
     } else {
       setAriaDescribedby('');
     }
+
+    let error = null;
+    if (isDirty || date) {
+      if (date) {
+        // show an error message right away
+        setIsDirty(true);
+      } else {
+        error = 'Please provide a valid date';
+      }
+      if (!nextPage) {
+        error = 'Please provide a valid future separation date';
```

Comment:
Changed this content based on [error message templates for dates](https://design.va.gov/patterns/forms/dates#error-message-templates-for-dates) - I will get a content review.